### DISCUSSION TITLE
pico core uses/disables DMA_IRQ_0 since 4.0.1 ?

### DISCUSSION THREAD

**Discussion by myklemykle** [earlephilhower/arduino-pico#2394] [Q&A] [answered]
Hi,

I just upgraded from arduino-pico 3.9.5 to 4.0.1 and my audio ISR stopped working. After some sleuthing I figured out that if I change it to trigger on DMA_IRQ_1 instead of DMA_IRQ_0, everything works as it should.

I see that the Pico SDK itself is under a lot of flux since the RP2350 was announced.  So I'm not sure what changes are due to evolution over there vs. in the arduino-pico package itself. But it makes me wonder what library or other thing is now using DMA in a different way than before?

I also wonder, is there any official list of already allocated resources (dma channels, pwm channels, irqs) that we should avoid using in our sketches?

Maybe there's some release notes I need to catch up on.

Thanks for any info, and again thanks very much for this awesome package.
-mykle-

**Comment by earlephilhower** [comment, marked as answer]
The core itself really has no changes to speak of, other than simple compatibility ones, for the 2.0.0 SDK update.  See https://github.com/earlephilhower/arduino-pico/pull/2336 .  The SDK internals, I really didn't dig into it because all of the DMAs and IRQs we use internally "just worked" with no changes.

I always use the `dma_claim_unused_channel` call to pick a DMA, not hardcode it, so even if they did pick one for something (which would surprise me but their code is out there to double check) it wouldn't matter because I'd automatically pick another one...

**Reply by myklemykle** [reply]
Thx. I use `dma_claim_unused_channel` too, but I believe there's only two available IRQs that a DMA channel can use to signal it's finished. Or I guess there's just one now? But one is enough for me.
Actually I will also check a couple arduino libraries that I upgraded at the same time, just with small point releases. I can't think of why any of them would suddenly develop a need to use DMA, but I guess it's not impossible.
